x-pack/plugins/cases/public/components/cases_context/index.tsx

```diff
@@ -6,40 +6,39 @@
  */
 
 import React, { useState, useEffect } from 'react';
+import { merge } from 'lodash';
+import { CasesContextValue, DEFAULT_FEATURES } from '../../../common';
 import { DEFAULT_BASE_PATH } from '../../common/navigation';
 import { useApplication } from './use_application';
 
-export interface CasesContextValue {
-  owner: string[];
-  appId: string;
-  appTitle: string;
-  userCanCrud: boolean;
-  basePath: string;
-}
-
 export const CasesContext = React.createContext<CasesContextValue | undefined>(undefined);
 
 export interface CasesContextProps
-  extends Omit<CasesContextValue, 'appId' | 'appTitle' | 'basePath'> {
+  extends Omit<CasesContextValue, 'appId' | 'appTitle' | 'basePath' | 'features'> {
   basePath?: string;
+  features?: Partial<CasesContextValue['features']>;
 }
 
-export interface CasesContextStateValue
-  extends Omit<CasesContextValue, 'appId' | 'appTitle' | 'userCanCrud'> {
+export interface CasesContextStateValue extends Omit<CasesContextValue, 'appId' | 'appTitle'> {
   appId?: string;
   appTitle?: string;
-  userCanCrud?: boolean;
 }
 
 export const CasesProvider: React.FC<{ value: CasesContextProps }> = ({
   children,
-  value: { owner, userCanCrud, basePath = DEFAULT_BASE_PATH },
+  value: { owner, userCanCrud, basePath = DEFAULT_BASE_PATH, features = {} },
 }) => {
   const { appId, appTitle } = useApplication();
+  /**
+   * The empty object at the beginning avoids the mutation
+   * of the DEFAULT_FEATURES object
+   */
+  const featuresOptions = merge({}, DEFAULT_FEATURES, features);
   const [value, setValue] = useState<CasesContextStateValue>({
     owner,
     userCanCrud,
     basePath,
+    features: featuresOptions,
```

Comment:
Nice! I would only suggest moving the initialization of the state inside a function so we prevent calling merge when it is not needed. Right now the merge result is only used the first render but it is computed every time, if we move it inside a function in the `useState` parameter React will only call it once:
```suggestion
  const [value, setValue] = useState<CasesContextStateValue>(() => ({
    owner,
    userCanCrud,
    basePath,
    features: merge({}, DEFAULT_FEATURES, features),
  }));
```

Reply:
I am not sure if that is correct. The JS engine will execute the `merge` function but React will not gonna reset the `value` after the first render. I change it because it is more readable :)

Reply:
> The JS engine will execute the merge function but React will not gonna reset the value after the first render
exactly https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
Though the last change did no differencem, because the `initialValue` parameter is still an object. The change I was suggesting is to replace the _object_ for a _function returning the object_, this way the merge will only be executed in the first render. 
Anyway, the merged `features` configs are pretty small by now, the optimization will not have a big impact. It is okay as it is 👍